src/classes/Collection.cls

```diff
@@ -2,317 +2,494 @@ public with sharing class Collection {
-		while (iter.hasNext()) {
+
+		while(iter.hasNext()) {
 			this.records.add(iter.next());
 		}
 	}
 
+
+	// CONVERT
+
 	public List<SObject> asList() {
 		return new List<SObject>(records);
 	}
 
+
 	public List<SObject> asList(Type listType) {
-		List<SObject> typedList = (List<SObject>) listType.newInstance();
+		List<SObject> typedList =(List<SObject>) listType.newInstance();
 		typedList.addAll(records);
 		return typedList;
 	}
 
+
 	public Set<SObject> asSet() {
 		return new Set<SObject>(records);
 	}
 
+
 	public Set<SObject> asSet(Type setType) {
-		Set<SObject> typedSet = (Set<SObject>) setType.newInstance();
+		Set<SObject> typedSet =(Set<SObject>) setType.newInstance();
 		typedSet.addAll(records);
 		return typedSet;
 	}
 
+
+	public Map<Id, SObject> asMap() {
+		return new Map<Id, SObject>(records);
+	}
+
+
+	public Map<Id, SObject> asMap(Type mapType) {
+		Map<Id, SObject> typedMap =(Map<Id, SObject>) mapType.newInstance();
+		typedMap.putAll(records);
+		return typedMap;
+	}
+
+
+	// FILTER
+
 	public Collection filter(SObjectPredicate predicate) {
-		List<SObject> filtered = new List<SObject>();
-		for (SObject record : records) {
-			if (predicate.apply(record)) {
-				filtered.add(record);
+		List<SObject> result = new List<SObject>();
+
+		for(SObject record : records) {
+			if(predicate.apply(record)) {
+				result.add(record);
 			}
 		}
-		return Collection.of(filtered);
+
+		return Collection.of(result);
 	}
 
+
 	public Collection remove(SObjectPredicate predicate) {
-		List<SObject> remaining = new List<SObject>();
-		for (SObject record : records) {
-			if (!predicate.apply(record)) {
-				remaining.add(record);
+		List<SObject> result = new List<SObject>();
+
+		for(SObject record : records) {
+			if(!predicate.apply(record)) {
+				result.add(record);
 			}
 		}
-		return Collection.of(remaining);
-	}
 
-	public List<Boolean> pluckBooleans(Schema.SObjectField field) {
-		List<Boolean> results = new List<Boolean>();
-		for (SObject rec : records) {
-			results.add((Boolean)rec.get(field));
-		}
-		return results;
+		return Collection.of(result);
 	}
 
-	public List<Boolean> pluckBooleans(String relation) {
-		List<Boolean> results = new List<Boolean>();
+
+	// EXTRACT
+
+	public List<Boolean> booleans(String fieldPath) {
+		List<Boolean> result = new List<Boolean>();
+
 		SObjectFieldReader reader = new SObjectFieldReader();
-		for (SObject rec : records) {
-			results.add((Boolean)reader.read(rec, relation));
+
+		for(SObject record : records) {
+			result.add((Boolean)reader.read(record, fieldPath));
 		}
-		return results;
+
+		return result;
 	}
 
-	public List<Date> pluckDates(Schema.SObjectField field) {
-		List<Date> results = new List<Date>();
-		for (SObject rec : records) {
-			results.add((Date)rec.get(field));
-		}
-		return results;
+
+	public List<Boolean> booleans(SObjectField field) {
+		return booleans(''+field);
+	}
+
+
+	public Set<Boolean> booleanSet(SObjectField field) {
```

Comment:
Originally, `Pluck.ids(collection)` would return a `Set<Id>` as a special case, because it's used in more often than any other `Id` field, and a `Set` is almost always more relevant. This was changed with the `Collection` rewrite, but could be reinstated by providing a `pluckIds` method without arguments which would default to a `Set` again: `pluckIds()` ~ `new Set<Id>(pluckIds('Id'))`. I believe that for everything not `Id` related, wrapping in client code will be sufficient because the use is rare, and it avoids the clutter. If it's really required, I urge changing the naming to original convention: pluck_Type_, and introducing pluck_Type_Set options.

Reply:
As discussed online. Revert my pluck changes.